### ISSUE TITLE
Unhandled Exception: DtlsException: DTLS Handshake has failed.

### ISSUE THREAD

**Issue by Ifilehk** [JKRhb/dtls2#77]
Bom dia again :-)

Not able to catch `DtlsException: DTLS Handshake` exception. Here the piece of code in `dtls_client.dart`

 ```
 void _performShutdown(Exception exception) {
    final wasInHandshake = inHandshake;
    close();

    if (wasInHandshake) {
      _connectCompleter.completeError(exception);
    } else {
      throw exception;
    }
  }
```

In fact `wasInHandshake` in false but event in this situation it should completeError the exception to be able to catch it upwards

**Comment by JKRhb** [comment]
Hi! ;) Could you check in which state the connection is when the exception is thrown?

**Comment by Ifilehk** [comment]
the state is `ConnectionState.closed`

**Comment by JKRhb** [comment]
Speaking of issues: Do you think this one is still relevant? Or has it been resolved by #75?

**Comment by Ifilehk** [comment]
Hem don't really remember the case when it happened.

For now and with the intensive use of your work, did not see anything pointing to that. So hopefully closed by #75

If something pointing to that will appears in future, I will reopen it. But for now could be closed.

**Comment by JKRhb** [comment]
Great, thank you for your feedback :)
